### PR TITLE
Automatically update NodeDB path if changed

### PR DESCRIPTION
**Pull request by yusufsahinhamza**
Fixes #3015

### CODE REVIEW

**Comment by nbougalis** [comment]
Thanks for your contribution! I’m going to ask @miguelportilla, @mtrippled and @jwbusch to take a look at this and sign off

**Comment by miguelportilla** [comment]
@yusufsahinhamza Considering narrowing the scope of work since it is only used in once place. Please take a look at [cabe1c41](https://github.com/miguelportilla/rippled/commit/cabe1c41743c9f630240ec7ccb48c66656ec5c8b)

**Comment by yusufsahinhamza** [comment]
@miguelportilla Yes, it's absolutely looks better now!

File: src/ripple/app/misc/SHAMapStoreImp.cpp [hunk 1]
```diff
@@ -554,6 +565,17 @@ SHAMapStoreImp::makeBackendRotating (std::string path)
     return backend;
 }
 
+std::string
+SHAMapStoreImp::updateDbPath(std::string path)
```
Comment (mtrippled) [inline src/ripple/app/misc/SHAMapStoreImp.cpp]: the path parameter should be const.

**Comment by mtrippled** [comment]
:+1:

**Pull request closed without merging**